src/lib.rs

```diff
@@ -110,6 +113,20 @@
 //! This will emit a metric like this: `metric.name:100|d|#tag:value`, note the metric type has
 //! emitted here is `d` and not `h`.
 //!
+//! **Note:** Most of the other metrics-rs builders provide a convenience method for installing a global recorder. E.g
+//! for Prometheus or TCP metrics exporters you could do something along the lines of `PrometheusBuilder::new().install()`.
+//! 
+//! This library does not provide such convenience method, the reason being that this is a independent exporter library
+//! and the version of `metrics` library that this library depends on might defer from the version you app uses. This 
+//! causes cargo to link both versions of the metrics crate, resulting in recorder installed in `metric-exporter-statsd` and
+//! the calls to `metrics::*` going through the version linked to your app. 
+//! 
+//! Having this library only return a recorder prevents this problem and also surfaces any version mismatch issues from upgrading
+//! one but not the other. 
+//! 
+//! 
+//! In addition, we align the version of this library with the version of metrics crate it uses. Hopefully that provides a somewhat 
```

Comment:
**Concern:** I don't think that we should align our version with upstream library because it makes it hard/impossible to properly do semantic versioning of **this** library.

Reply:
Well, here's the situation:

- When `metrics` releases a breaking version, we **must** do the same to avoid all users unwittingly having all their data routed to /dev/null

- Users have to somehow figure out which version to use

- All the other `metrics-recorder-*` crates version in lockstep with `metrics`

- We are fairly unlikely (it's not impossible) to want to publish a breaking version of our own for any other reason excpet to keep up with `metrics`

I think we should do this, even though you're right it's not exactly semver

Reply:
If `metrics` ever goes to 1.0 and settles down on a stable API, we can re-evaluate.

Reply:
We decided to let this library have it's own versioning instead of coupling with metrics because we may want to add features to this lib that may not directly tie to the version of metrics lib.

Reply:
Hey there!

Discovered this crate on a whim, and was digging through the activity, and thought I could add some color here. :)

Generally speaking, my time to work on `metrics` has contracted a bit given work obligations, etc. I've been trying to continue shrinking the size of API changes, or at least remove things where possible to make the surface smaller. I'm unofficially aiming for a v1 release in a few months after I can decide on, and implement, solutions for some outstanding sharp edges:

- the trade-off between the `Recorder` seeing every metric operation vs needing to provide handles that can do so
- inflexibility around performant tagging (namely that we could do a better job i.e. supporting `Arc<str>` or inlined strings a la `smartstring`)
- whether or not there needs to be another core data type for rates/meters or if that could be reasonably derived from existing data on the exporter side
- in the same vein, does the current definition of a histogram in `metrics` provide sufficient flexibility for the way that metrics system X uses "histograms"? (the code y'all have for distinguishing histos vs timers vs ... is a perfect example of the impedance mismatch that can occur between `metrics` and real-life)

I'm always interested in hearing from users about their experiences both using `metrics` and writing exporters/integrations for it, so if there's something on your minds, don't hesitate to file an issue on the repository or come chat with me in [Discord](https://discord.gg/eTwKyY9).